dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java

```diff
@@ -174,9 +174,22 @@ public Optional<SourceFile> readFile(String repositoryId, Version<?> tag, Descri
     public abstract Map<String, String> getWorkflowGitUrl2RepositoryId();
 
     /**
-     * Checks to see if a particular source code repository is properly setup for issues like token scope
+     * Checks to see if a particular source code repository is properly setup for issues like token scope.
      */
-    public abstract boolean checkSourceCodeValidity();
+    public abstract boolean checkSourceControlTokenValidity();
+
+    /**
+     * Checks to see if a particular source code repository is real or not.
+     * TODO: assumes real repos have branches, there may be a better way for specific source control types
+     *
+     * @param entry
+     * @return
+     */
+    public boolean checkSourceControlRepoValidity(Entry entry) {
+        final String repositoryId = this.getRepositoryId(entry);
+        final String mainBranch = this.getMainBranch(entry, repositoryId);
```

Comment:
Does this stop you from registering a tool with a GitHub repo that you don't have permissions to?

If not, it's beyond the scope of this ticket, and legacy tools are different anyway, since the namespace is controlled by the container. But you mentioned #5094 in the PR description, so I just wanted to make sure this is the intent.

Reply:
No, it does not. I just meant that the structure of the solution and tests ends up being similar although this checks for invalid repos and that checks for access to the repo in question. 

I think a follow-up for checking whether you have access to the tool source code in question would be possible as well. 
That said, apptools would get this for free.